src/libraries/System.Net.Http/tests/UnitTests/Headers/HttpHeadersTest.cs

```diff
@@ -1646,6 +1647,177 @@ public void GetEnumerator_UseExplicitInterfaceImplementation_EnumeratorReturnsNo
             Assert.False(enumerator.MoveNext(), "Only 2 values expected, but enumerator returns a third one.");
         }
 
+        [Fact]
+        public void NonValidated_GetEnumeratorFromUninitializedHeaderStore_ReturnsEmptyEnumerator()
+        {
+            MockHeaders headers = new MockHeaders();
+
+            var enumerator = headers.NonValidated.GetEnumerator();
+            Assert.False(enumerator.MoveNext());
+        }
+
+        [Fact]
+        public void NonValidated_FirstHeaderWithOneValueSecondHeaderWithTwoValues_EnumeratorReturnsTwoHeaders()
```

Comment:
Can we add a test that does something like this:

(1) Add raw header via TryAddWithoutValidation
(2) Retrieve it, causing it to be parsed
(3) Use NonValidatedHeaders and confirm that it's still returning the original raw value, unmodified

Reply:
Added [NonValidated_AddRawHeaderValueThenRetrieveItAsParsed_EnumeratorReturnsOriginalRawValue](https://github.com/dotnet/runtime/blob/9c860a47d50cedbbe60f49ef891cd0b453bb0e9e/src/libraries/System.Net.Http/tests/UnitTests/Headers/HttpHeadersTest.cs#L1907) test.